### ISSUE TITLE
Feature Request: add `write_datetime_with_format`, while `write_datetime` should not receive a format

### ISSUE THREAD

**Issue by lucatrv** [jmcnamara/rust_xlsxwriter#47]
### Feature Request

I set a column format with 

```rust
worksheet.set_column_format(col, &rust_xlsxwriter::Format::new().set_num_format("dd/mm/yyyy"))?;
```

So IMHO I should be able to write dates to that column with:

```rust
worksheet.write_datetime(row, col, &date)?;
```

instead of having to repeat the same format with:

```rust
worksheet.write_datetime(row, col, &date, &rust_xlsxwriter::Format::new().set_num_format("dd/mm/yyyy"))?;
```

In other words, the plain `write_datetime` method should not receive a format, while a separate `write_datetime_with_format` method should be added for those cases where the date format has not already been applied to the cell.

**Comment by jmcnamara** [comment]
> So IMHO I should be able to write dates to that column with:

You are correct. You should be able to do that.

I have implemented `write_datetime()` as `write_datetime_with_format()` since it almost never makes sense in Excel to write a date without a format. However, that creates an inconsistency for the case you pointed out so I will need to fix.

Thanks for the input.

**Comment by jmcnamara** [comment]
This is now fixed on main and will be in the next release.

The `write_datetime()` method no longer takes a format (that becomes `write_datetime_with_format()`) and the `write()` method no longer add a default date/time format.

The datetime will pick up any implicit column or row formatting or display as an unformatted number.

**Comment by jmcnamara** [comment]
This is fixed in version v0.42.0. Thanks for the report.
